sdks/python/apache_beam/typehints/typehints_test.py

```diff
@@ -612,54 +613,70 @@ def test_match_type_variables(self):
                      hint.match_type_variables(typehints.Dict[int, str]))
 
 
-class SetHintTestCase(TypeHintTestCase):
+class BaseSetHintTest:
```

Comment:
Just pushed a fix - BaseSetHintTest is now a wrapper for a nested class that inherits from TypeHintTestCase (which inherits from TestCase), removing the multiple inheritance.

I think BaseSetHintTest can't directly inherit from TypeHintTestCase because that would cause the unittest library to accidentally run the BaseSetHintTest tests which cannot be run without first being instantiated by the children classes (SetHintTestCase, FrozenSetHintTestCase). The nested class solves this issue. See [here](https://stackoverflow.com/a/25695512).

Reply:
Just pushed a fix - BaseSetHintTest is now a wrapper for a nested class that inherits from TypeHintTestCase (which inherits from TestCase), removing the multiple inheritance.

I think BaseSetHintTest can't directly inherit from TypeHintTestCase because that would cause the unittest library to accidentally run the BaseSetHintTest tests which cannot be run without first being instantiated by the children classes (SetHintTestCase, FrozenSetHintTestCase). The nested class solves this issue. See [here](https://stackoverflow.com/a/25695512).

Reply:
Nice solution! I've run into this in the past.

What's happening is that pytest is loading all modules and inspecting their attributes looking for tests to run. Another solution solution would have been to set `BaseSetHintTest = None` near the bottom of this module. Effectively both solutions are hiding the base class from being collected by pytest.

Reply:
That makes sense, thanks for the explanation!